Aptoide/aptoide-client-v8
# Feature/an 1684 timeline with card

What does this PR do?

   This pull-request makes it possible to navigate to Timeline passing a cardId so it refreshes with that cardId in the timeline request body.
   The main goal of the ticket was: After making a new Post, user goes back to the Timeline with the card on the top of the list.

   Changes were made to the Post response model so we get the cardId from the Post response body request and then the TabNavigator is called to navigate to the Timeline with cardId.
   TabNavigator was changed to use a PublishRelay instead of Behavior so we could control the chaching of navigation events. There is now a new method in the TabNavigator interface to clear the navigation cache that is used in the Timeline so it doesn't navigate on subscribe when view is re-created.


Where should the reviewer start?

- [ ] TimelineFragment
- [ ] TimelinePresenter
- [ ] PostPresenter

How should this be manually tested?

  Open Aptoide v8 > Timeline > click on FAB > write text in edit text & select app > click post > Timeline should open with new request and Post on top. (If the backend is slow the request might not come with the just Posted Post, verify charles request if it has a cardId in the request body (getUserTimeline))

What are the relevant tickets?

Tickets related to this pull-request: [AN-1684](https://aptoide.atlassian.net/browse/AN-1684)

Screenshots (if appropriate):
N/A

Questions:

   Is there a blog post? N/A
   Does this add new dependencies which need to be added? No
